source-facebook-marketing/source_facebook_marketing/source_facebook_marketing/spec.py

```diff
@@ -0,0 +1,235 @@
+        title="Breakdowns",
+        description="A list of chosen breakdowns for breakdowns",
+        default=[],
+    )
+
+    action_breakdowns: Optional[List[ValidActionBreakdowns]] = Field(
+        title="Action Breakdowns",
+        description="A list of chosen action_breakdowns for action_breakdowns",
+        default=[],
+    )
+
+    time_increment: Optional[PositiveInt] = Field(
+        title="Time Increment",
+        description=(
+            "Time window in days by which to aggregate statistics. The sync will be chunked into N day intervals, where N is the number of days you specified. "
+            "For example, if you set this value to 7, then all statistics will be reported as 7-day aggregates by starting from the start_date. If the start and end dates are October 1st and October 30th, then the connector will output 5 records: 01 - 06, 07 - 13, 14 - 20, 21 - 27, and 28 - 30 (3 days only)."
+        ),
+        exclusiveMaximum=90,
+        default=1,
+    )
+
+    start_date: Optional[datetime] = Field(
+        title="Start Date",
+        description="The date from which you'd like to replicate data for this stream, in the format YYYY-MM-DDT00:00:00Z.",
+        pattern=DATE_TIME_PATTERN,
+        examples=["2017-01-25T00:00:00Z"],
+    )
+
+    end_date: Optional[datetime] = Field(
+        title="End Date",
+        description=(
+            "The date until which you'd like to replicate data for this stream, in the format YYYY-MM-DDT00:00:00Z. "
+            "All data generated between the start date and this end date will be replicated. "
+            "Not setting this option will result in always syncing the latest data."
+        ),
+        pattern=DATE_TIME_PATTERN,
+        examples=["2017-01-26T00:00:00Z"],
+    )
+    insights_lookback_window: Optional[PositiveInt] = Field(
+        title="Custom Insights Lookback Window",
+        description="The attribution window",
+        # maximum=28,
+        # mininum=1,
+        default=28,
+    )
+
+class AccessToken(BaseModel):
+    access_token: str = Field(
+        title="Access Token",
+        order=1,
+        description=(
+            "The value of the generated access token. "
+            'From your App’s Dashboard, click on "Marketing API" then "Tools". '
+            'Select permissions <b>ads_management, ads_read, read_insights, business_management</b>. Then click on "Get token". '
+            'See the <a href="https://docs.airbyte.com/integrations/sources/facebook-marketing">docs</a> for more information.'
+        ),
+        airbyte_secret=True,
+    )
+
+class OAuthCredentials(BaseModel):
+    access_token: str = Field(
+        airbyte_secret=True
+    )
+
+    auth_type: str = Field(
+        default="OAuth Credentials",
+        order=0,
+    )
+    client_id: str = Field(
+        airbyte_secret=True
+    )
+    client_secret: str = Field(
+        airbyte_secret=True
+    )
+
+    class Config:
```

Comment:
This is how you configure these Pydantic 1.x dataclasses to have the `x-oauth2-provider` field. Took me a while to figure this out, 1.x docs are much worse compared to 2.x docs